src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java

```diff
@@ -0,0 +1,266 @@
+    for (String variantFile : variantFileNames) {
+      String variantFileAbsPath = new File(variantFile).getAbsolutePath();
+      AbstractFeatureReader<VariantContext, LineIterator> reader =
+        AbstractFeatureReader.getFeatureReader(variantFileAbsPath, new VCFCodec(), true);
+      headers.add((VCFHeader)reader.getHeader());
+
+      // We assume only one sample per file
+      String sampleName = ((VCFHeader) reader.getHeader()).getGenotypeSamples().get(0);
+      sampleToVCFMap.put(sampleName, reader);
+    }
+
+    Set<VCFHeaderLine> mergedHeader = VCFUtils.smartMergeHeaders(headers, true);
+    sizePerColumnPartitionPerSample *= sampleToVCFMap.size();
+
+    logger.info("Writing data to array - " + workspace + "/" + arrayName);
+
+    try {
+      importer = new GenomicsDBImporter(sampleToVCFMap, mergedHeader,
+        intervals.get(0), workspace, arrayName,
+        sizePerColumnPartitionPerSample, segmentSize,
+        vidMapJSONFile, callsetMapJSONFile);
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * A complete traversal from start to finish. Tool authors who wish to "roll their own" traversal
+   * from scratch can extend this class directly and implement this method. Walker authors should
+   * instead extend a Walker class and implement the Walker-appropriate apply() method, since the
+   * Walker base classes implement the various kinds of traversals for you.
+   */
+  @Override
+  public void traverse() {
+    try {
+      importer.importBatch();
+      logger.info("Current batch written to disk");
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Loads our intervals using the best available sequence
+   * dictionary (as returned by {@link #getBestAvailableSequenceDictionary})
+   * to parse/verify them. Does nothing if no intervals were specified.
+   */
+  private void initializeIntervals() {
+    if ( intervalArgumentCollection.intervalsSpecified() ) {
+      final SAMSequenceDictionary intervalDictionary = getBestAvailableSequenceDictionary();
+      if ( intervalDictionary == null ) {
+        throw new UserException("We require at least one input source that " +
+          "has a sequence dictionary (reference or reads) when intervals are specified");
+      }
+
+      intervals = new ArrayList<>();
+
+      List<SimpleInterval> simpleIntervalList =
+        intervalArgumentCollection.getIntervals(intervalDictionary);
+
+      for (SimpleInterval simpleInterval : simpleIntervalList) {
+        intervals.add(new ChromosomeInterval(simpleInterval.getContig(),
+          simpleInterval.getStart(), simpleInterval.getEnd()));
+      }
+    }
+  }
+
+  /**
+   * Close the GenomicsDB importer
+   * once all variants are written
+   */
+  @Override
+  public void closeTool() {
+    assert importer.isDone();
+    logger.info("Successfully imported " + variantFileNames.size() + " callsets");
```

Comment:
Move this success message to an override of `onTraversalSuccess()`. Override `closeTool()` only if you need to close resources (like readers) at the end of traversal.